### PR TITLE
feat: confidence-weighted retrieval ranking (#181)

### LINKED ISSUE #181: feat(memory): add confidence-weighted fact selection and priority display to memory loading
## Problem

`MemoryContextLoader` loads up to 100 facts ordered by date with no quality filtering. `SessionContextFormatter` displays recent memories as a flat time-grouped list with no confidence indicators. As facts accumulate at varying quality levels (verified decisions, low-confidence auto-inferred, ephemeral prompts), the context becomes noisy. The memory-strategy.md (Phase C) calls for a ranking algorithm that selects the most valuable ~30 facts using confidence/source priority tiers and formats them with confidence indicators.

## Proposed Solution

Add a tier-based ranking algorithm to `MemoryContextLoader` and priority-grouped display to `SessionContextFormatter`.

| File | Action | Details |
|------|--------|---------|
| `src/lib/application/services/MemoryContextLoader.ts` | Modify | Add `rankAndSelectFacts()` with 6-tier priority algorithm, budget cap of 30, file-aware boost; reduce raw fetch from 100 to 60 |
| `src/lib/application/services/SessionContextFormatter.ts` | Modify | Add confidence indicators `[V]`/`[H]`/`[M]`, priority grouping (Key facts / Recent context / Background), per-tier truncation |
| `tests/unit/.../MemoryContextLoader.ranking.test.ts` | Create | ~20 test cases for tier assignment, budget cap, file boost, ordering |
| `tests/unit/.../SessionContextFormatter.priority.test.ts` | Create | ~12 test cases for indicators, grouping, truncation, backward compat |

### Ranking tiers

| Tier | Criteria | Cap |
|------|----------|-----|
| 1 | `confidence:verified` | Always included |
| 2 | `source:user-explicit` | Always included |
| 3 | `confidence:high` + created within 48h | Up to 15 |
| 4 | Active tasks (`type:task`, status not `done`) | Up to 10 |
| 5 | `confidence:medium` + created within 24h | Up to 5 |
| 6 | Everything else, by date | Fills remaining budget |

Total budget: 30 facts. Fill from Tier 1 downward.

### Display format

- `[V]` verified, `[H]` high, `[M]` medium, no indicator for low/uncertain
- Grouped: "Key facts" (T1-2), "Recent context" (T3-5), "Background" (T6)
- Per-tier truncation at 10 items with "... and N more"
- `ranked: false` flag preserves existing time-grouped format (backward compat)

## Acceptance Criteria

- [ ] `rankAndSelectFacts()` correctly implements 6-tier priority algorithm
- [ ] Budget cap of 30 facts enforced across all tiers
- [ ] Tier assignment uses `parseConfidenceTag()` and `parseSourceTag()` from existing quality utilities
- [ ] File-aware query boosts facts matching `recentFilePaths` parameter
- [ ] `SessionContextFormatter` displays `[V]`, `[H]`, `[M]` confidence indicators
- [ ] Priority grouping: "Key facts", "Recent context", "Background"
- [ ] Per-tier truncation at 10 items with "+N more" indicator
- [ ] Backward compatible — `ranked: false` preserves existing display format
- [ ] All existing tests pass without modification
- [ ] New code covered by unit tests (~32 test cases)
- [ ] All tests pass (`npm test`)
- [ ] Linting passes (`npm run lint`)
- [ ] TypeScript compiles (`npx tsc --noEmit`)

## References

- Part of #178
- Depends on PR B (#180)
- `.dev/features/memory/memory-strategy.md` (Phase C: Retrieval Ranking, Section 2)
- `.dev/features/memory/memory-strategy-execution-plan.md` (Issue 3 / PR C)
- `src/lib/application/services/MemoryContextLoader.ts`
- `src/lib/application/services/SessionContextFormatter.ts`
- `src/lib/domain/interfaces/types/IMemoryQuality.ts`

### PR DESCRIPTION
**Pull request by TonyCasey**
## Summary

- Add 6-tier priority ranking algorithm to `MemoryContextLoader.rankAndSelectFacts()` with per-tier budget caps (T1-T2 unlimited, T3 max 15, T4 max 10, T5 max 5, total 30)
- Add confidence indicators (`[V]`/`[H]`/`[M]`) and priority-grouped display (Key facts / Recent context / Background) to `SessionContextFormatter`
- File-aware boost promotes facts mentioning recent file paths to Tier 3
- Backward compatible: `ranked=false` or omitted uses existing time-grouped format

## Test plan

- [x] 17 new ranking tests (tier assignment, budget caps, file-aware boost, ordering, edge cases)
- [x] 20 new priority display tests (indicators, tiers, group formatting, backward compatibility)
- [x] All 1456 existing tests pass
- [x] Lint clean
- [x] Build succeeds

Closes #181

🤖 Generated with [Claude Code](https://claude.com/claude-code)

## Summary by CodeRabbit

* **New Features**
  * 6-tier confidence-weighted retrieval ranking with per-tier caps and file-aware boosts.
  * Priority-based memory display grouped into Key facts, Recent context, and Background; toggleable ranked view with confidence indicators ([V]/[H]/[M]).
  * Task-type detection for mode-aware memory retrieval and decision signaling.
  * Ranked format is optional; existing time-based display remains supported.

* **Tests**
  * Comprehensive unit tests for ranking, priority display, and tier computation.

### CODE REVIEW

File: tests/unit/src/lib/domain/interfaces/types/IMemoryQuality.tier.test.ts [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+
+  it('returns Tier 5 for medium confidence within 24h', () => {
+    const recent = new Date(Date.now() - 3 * 60 * 60 * 1000).toISOString();
+    const item = createItem({ tags: ['confidence:medium'], created_at: recent });
+    assert.strictEqual(computeMemoryTier(item), 5);
+  });
+
+  it('returns Tier 6 for everything else', () => {
+    const item = createItem({ tags: ['confidence:low'] });
+    assert.strictEqual(computeMemoryTier(item), 6);
```
[the remark this answers is not in the document]
Comment (TonyCasey) [reply tests/unit/src/lib/domain/interfaces/types/IMemoryQuality.tier.test.ts]: Renamed computeMemoryTier test cases to use the "should …" naming convention.
